### PR TITLE
Updated MAUI Sample Viewer to NET 7

### PR DESCRIPTION
**Pull request by duffh**
# Description

Updated MAUI Sample Viewer to NET 7 and fixed issues encountered during the update. 

## Type of change

- [ ] Bug fix
- [ ] New sample implementation
- [x] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [x] MAUI Android
- [x] MAUI Windows
- [ ] MAUI Mac Catalyst
- [ ] MAUI iOS

## Checklist

- [x] Runs and compiles on all active platforms
- [x] Branch is up to date with the latest main/v.next
- [x] All merge conflicts have been resolved
- [x] Self-review of changes
- [x] There are no warnings related to changes
- [x] No unrelated changes have been made to any other code or project files
- [ ] `sample_sync.py` runs without making changes
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] Code is commented with correct formatting
- [x] All variable and method names are good and make sense
- [x] There is no leftover commented code

### CODE REVIEW

File: src/MAUI/Maui.Samples/SettingsPage.xaml [hunk 1]
```diff
@@ -37,10 +37,10 @@
                                 Clicked="DeleteAllClicked"
                                 IsEnabled="{x:OnPlatform Android=True,
```
Comment (ZackAllen) [inline src/MAUI/Maui.Samples/SettingsPage.xaml:38]: If we enable this everywhere, shouldnt it just be set as true?

**Comment by amweis226** [comment]
We should change any mentions of NET 6 to NET 7 i.e. reference doc tag on the readme page, any links, etc.

File: src/MAUI/Maui.Samples/SettingsPage.xaml [hunk 2]
```diff
@@ -90,10 +86,10 @@
                                                      HeightRequest="25"
                                                      IsEnabled="{x:OnPlatform Android=True,
```
Comment (ZackAllen) [inline src/MAUI/Maui.Samples/SettingsPage.xaml:87]: We don't need OnPlatform if we do the same value on each platform.
It should be enabled and visible everywhere by default, we can remove the setters entirely.

**Review by ZackAllen — APPROVED** [review]
Approved with the OnPlatform change and Andys comment resolved.

**Comment by duffh** [comment]
> We should change any mentions of NET 6 to NET 7 i.e. reference doc tag on the readme page, any links, etc. Looks good to me otherwise

I've searched for net6 + net 6 and haven't come back with any hits on the MAUI viewer so I think it's all good to go.

**Pull request merged**